### ISSUE TITLE
[Google Drive] KeyError('NID')

### ISSUE THREAD

**Issue by vagabondshrimp** [yt-dlp/yt-dlp#6909]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Other links for testing:
- [One](https://drive.google.com/file/d/1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp/view?usp=share_link)
- [Two](https://drive.google.com/file/d/1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0/view?usp=share_link)
- [Three](https://drive.google.com/file/d/1z06lRE5n2tdhO_UoxK66add4UjdZIBKn/view?usp=share_link)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU "https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link"
[debug] Command-line config: ['-vU', 'https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link']
[debug] User config "C:\Users\config.txt": ['-4', '--geo-bypass', '--no-mtime', '--cookies', 'C:\\Users\\cookies.txt', '-S', 'fps,tbr', '--downloader', 'aria2c', '--downloader-args', 'aria2c:-c -x 10 -s 10', '--add-metadata', '--parse-metadata', ':(?P<meta_comment>)(?P<meta_description>)(?P<meta_synopsis>)', '--embed-subs', '--sub-lang', 'en,en-US', '--convert-thumbnails', 'jpg', '--embed-thumbnail', '-o', 'C:\\Users\\Downloads\\Video\\%(title)s.%(ext)s', '-f', 'bv*[height<=720]+ba/bv*[height<=720]/b', '--remux-video', 'mkv']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.3 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: ffmpeg 2023-04-03-git-6941788d24-essentials_build-www.gyan.dev (setts)
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.2
[debug] Proxy map: {}
[debug] Loaded 1805 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[GoogleDrive] Extracting URL: https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Downloading video webpage
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Requesting source file
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Requesting confirmed source file
ERROR: 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: An extractor error has occurred. (caused by KeyError('NID')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\common.py", line 694, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\googledrive.py", line 271, in _real_extract
    self.cookiejar.clear(domain='.google.com', path='/', name='NID')
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\http\cookiejar.py", line 1708, in clear
    del self._cookies[domain][path][name]
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: 'NID'
```

**Comment by observeroftime01** [comment]
Your URL from the verbose log appears to work just fine:

<details>
<summary>The URL from your log</summary>

```
yt-dlp.exe -vU -f- https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link
[debug] Command-line config: ['-vU', '-f-', 'https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev (setts), ffprobe 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: NSigDeno (YoutubeIE), Youtube_NsigDenoIE
[debug] Loaded 1787 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[GoogleDrive] Extracting URL: https://drive.google.com/file/d/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW/view?usp=share_link
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Downloading video webpage
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Requesting source file
[GoogleDrive] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Requesting confirmed source file
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW:
ID     EXT  RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
18     mp4  640x360    │ https │ unknown unknown
22     mp4  1280x720   │ https │ unknown unknown
37     mp4  1920x1080  │ https │ unknown unknown
source webm unknown    │ https │ unknown unknown

Enter format selector: source
[info] 1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW: Downloading 1 format(s): source
[debug] Invoking http downloader on "https://doc-0k-7o-docs.googleusercontent.com/docs/securesc/g4gu8r7q9l275vub4d44abm5tqlio4at/ha4fagenul015fga9k1096epgvjk7en4/1682455125000/17951362124911303423/06748896429223469761Z/1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW?e=download&uuid=b6b07a7a-2c08-4e30-af4d-025bd867868a&nonce=8jp3v2igbsvdo&user=06748896429223469761Z&hash=rp95l05belf2f60rvvsifsrkb6v2lqk0"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: 20200107 - uHQPafkPoYw - FORSEN PLAYS GTA - SAN ANDREAS.webm [1onfonY1Xrfi4hWOojh8VHjNWKRUBF3RW].webm
[download] 100% of  405.47MiB in 00:00:14 at 27.46MiB/s
```

</details>

<details>
<summary>One</summary>

```
yt-dlp.exe -vU -F https://drive.google.com/file/d/1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp/view?usp=share_link
[debug] Command-line config: ['-vU', '-F', 'https://drive.google.com/file/d/1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp/view?usp=share_link']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev (setts), ffprobe 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: NSigDeno (YoutubeIE), Youtube_NsigDenoIE
[debug] Loaded 1787 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[GoogleDrive] Extracting URL: https://drive.google.com/file/d/1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp/view?usp=share_link
[GoogleDrive] 1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp: Downloading video webpage
[GoogleDrive] 1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp: Requesting source file
[GoogleDrive] 1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp: Requesting confirmed source file
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1erZgnqKZz6ruVKOcHOwA9Y5aaAEGFrSp:
ID     EXT  RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
18     mp4  640x360    │ https │ unknown unknown
22     mp4  1280x720   │ https │ unknown unknown
37     mp4  1920x1080  │ https │ unknown unknown
source webm unknown    │ https │ unknown unknown
```

</details>


<details>
<summary>Two</summary>

```
yt-dlp.exe -vU -F https://drive.google.com/file/d/1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0/view?usp=share_link
[debug] Command-line config: ['-vU', '-F', 'https://drive.google.com/file/d/1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0/view?usp=share_link']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev (setts), ffprobe 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: NSigDeno (YoutubeIE), Youtube_NsigDenoIE
[debug] Loaded 1787 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[GoogleDrive] Extracting URL: https://drive.google.com/file/d/1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0/view?usp=share_link
[GoogleDrive] 1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0: Downloading video webpage
[GoogleDrive] 1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0: Requesting source file
[GoogleDrive] 1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0: Requesting confirmed source file
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1NpATxoiFsOl8tETkuZ59oP4HOSy2b8W0:
ID     EXT  RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
18     mp4  640x360    │ https │ unknown unknown
22     mp4  1280x720   │ https │ unknown unknown
37     mp4  1920x1080  │ https │ unknown unknown
source webm unknown    │ https │ unknown unknown
```

</details>

<details>
<summary>Three</summary>

```
yt-dlp.exe -vU -F https://drive.google.com/file/d/1z06lRE5n2tdhO_UoxK66add4UjdZIBKn/view?usp=share_link
[debug] Command-line config: ['-vU', '-F', 'https://drive.google.com/file/d/1z06lRE5n2tdhO_UoxK66add4UjdZIBKn/view?usp=share_link']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev (setts), ffprobe 2023-01-22-git-9d5e66942c-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Extractor Plugins: NSigDeno (YoutubeIE), Youtube_NsigDenoIE
[debug] Loaded 1787 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[GoogleDrive] Extracting URL: https://drive.google.com/file/d/1z06lRE5n2tdhO_UoxK66add4UjdZIBKn/view?usp=share_link
[GoogleDrive] 1z06lRE5n2tdhO_UoxK66add4UjdZIBKn: Downloading video webpage
[GoogleDrive] 1z06lRE5n2tdhO_UoxK66add4UjdZIBKn: Requesting source file
[GoogleDrive] 1z06lRE5n2tdhO_UoxK66add4UjdZIBKn: Requesting confirmed source file
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1z06lRE5n2tdhO_UoxK66add4UjdZIBKn:
ID     EXT  RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
18     mp4  640x360    │ https │ unknown unknown
22     mp4  1280x720   │ https │ unknown unknown
37     mp4  1920x1080  │ https │ unknown unknown
source webm unknown    │ https │ unknown unknown
```

</details>

Perhaps there's an issue with your config somehow. Try it with the defaults, i.e `--ignore-config` and see if you still have the same issues. Maybe there's an issue with using your own cookies and the option `--geo-bypass`, seeing how the GoogleDrive extractor has an issue with cookies from the looks of it.

**Comment by vagabondshrimp** [comment]
> Maybe there's an issue with using your own cookies and the option --geo-bypass, seeing how the GoogleDrive extractor has an issue with cookies from the looks of it.

Yep, this is it. Tried --cookies-from-browser, and that fixed it.
Thanks for the help.
